met-proxy.py

```diff
@@ -331,71 +331,55 @@ async def getHolisticResponse(lat: float, lon: float, userAgent: str, locationIq
-
-        locationIqKey = ""
-        try:
-            locationIqKey = qs['locationIqApiKey'][0]
-        except KeyError:
-            pass
-
-        return lat, lon, locationIqKey
-
-    def do_GET(self):
-        userAgent = userAgentDefault
-        if allowOverrideUserAgent:
-            userAgent = self.headers.get("User-Agent", userAgentDefault)
+def parseRequest(qs):
+    if 'lat' not in qs or 'lon' not in qs:
+        printColor(qs, color=bcolors.RED)
+        raise web.HTTPBadRequest(reason=
+            "Error: expecting GET request with lat and lon parameters")
+
+    lat, lon = float(qs['lat']), float(qs['lon'])
+    if lat < -90 or lat > 90 or lon < -180 or lon > 180:
+        raise web.HTTPBadRequest(reason=
+            "Error: lat and lon must be between -90 and 90 and -180 and 180 respectively")
+
+    locationIqKey = qs.get("locationIqApiKey", "")
+
+    return lat, lon, locationIqKey
+
+
+async def handleRequest(request: web.Request) -> web.Response:
+    userAgent = userAgentDefault
+    if allowOverrideUserAgent:
+        userAgent = request.headers.get("User-Agent", userAgentDefault)
+
+    printColor(
+        f" - Serving Incoming request {bcolors.BOLD}{request.path_qs} with User-Agent: {userAgent}",
+        color=bcolors.PURPLE)
+
+    try:
+        lat, lon, locationIqApiKey = parseRequest(request.query)
+
+        with cacheLock:
+            dataExpires, data = cache.get((lat, lon), (None, None))
+
+        if data is None or dataExpires < time.time():
+            data = await getHolisticResponse(lat, lon, userAgent, locationIqApiKey)
+        else:
+            printColor(f"Serving cached data for {request.path_qs} - cache expires in {int(dataExpires - time.time())} seconds", color=bcolors.YELLOW)
+
         printColor(
-            f" - Serving Incoming request {bcolors.BOLD}{self.path} with User-Agent: {userAgent}",
-            color=bcolors.PURPLE)
-
-        try:
-            lat, lon, locationIqApiKey = self.parseIncomingRequest()
-            with cacheLock:
-                dataExpires, data = cache.get((lat, lon), (None, None))
-            if data is None or dataExpires < time.time():
-                data = eventLoop.run_until_complete(getHolisticResponse(lat, lon, userAgent, locationIqApiKey))
-            else:
-                printColor(f"Serving cached data for {self.path} - cache expires in {int(dataExpires - time.time())} seconds", color=bcolors.YELLOW)
-
-            self.send_response(200)
-            self.send_header("Content-type", "application/json")
-            self.end_headers()
-            printColor(
-                f" - Serving data for {self.path}", color=bcolors.BOLD + bcolors.GREEN)
-            if debug:
-                pprint(json.loads(data), compact=True)
-                sys.stdout.flush()
-
-            self.wfile.write(data)
-        except BrokenPipeError:
-            printColor(
```

Comment:
Unless aiohttp handles this gracefully, you probably want to handle the BrokenPipe explicitly - that's when the client disconnect and you cannot respond to the client at all (not even with an error response). I remember I was getting an exception when handling the BaseException.

Reply:
I think aiohttp should handle it, as we do not explicitly write to the socket ourselves anymore with `self.wfile.write(data)` (which is probably where the exception got raised?), but rather just give the data to aiohttp by returning the `web.Response`.

Do you have a way to reproduce it to test? I tried triggering it with `curl --max-time 0.001 "http://localhost:8080/?lat=60.1234&lon=9.623514"`, but didn't get it even with the old code.